### ISSUE TITLE
CParser packed attribute parsing seems to not properly work in structs and enums

### ISSUE THREAD

**Issue by antoniovazquezblanco** [NationalSecurityAgency/ghidra#6776]
**Describe the bug**
I've got the following C source for an Xtensa:LE:32:default target:
```c
typedef union
{
    uint16_t raw;
    struct __attribute__((packed))
    {
        unsigned field1 : 15;
        unsigned field2: 1;
    } fields;
} mystruct_t;
static_assert(sizeof(mystruct_t) == 2);
```
This is compiled without warnings nor errors by the default IDF toolchain for the architecture. This implies that the size of the struct is 2 bytes.

If I parse the source code with ghidra, this results in the following:
![imagen](https://github.com/user-attachments/assets/3f8556fa-d33b-447a-854b-c152dbc784ad)

This produces a 4 byte sized structure instead of the desired 2 bytes...
Manually defining a custom packing solves the issue and reduces the size of the structure.

**Environment (please complete the following information):**
 - OS: Windows 11
 - Java Version: 21.0.3
 - Ghidra Version: 11.1.1
 - Ghidra Origin: Official GitHub distro

**Comment by Wall-AF** [comment]
This is occurring because the default alignment for your Language ID is 4 bytes.  I had this when Ghidra defaults to using datatypes from `windows_vs12_32.gdt` instead of one I've prepared for `MSVC_4.20.gdt`.  The only way I found was to manually update the `align` to (in this case) `2` instead of the `default` for all components of your structures also (in this case, none) and/or replace the datatypes with those from my `gdt`.

**Comment by antoniovazquezblanco** [comment]
I am quite confused by this...

Following what you said @Wall-AF I made a couple of tests...

1. Regarding the comment that Ghidra defaults to clib/windows_vs datatypes I made sure my structs only used builtin types. In this case unsigned is builtin and should respect my platform settings. This means that this should not depend on how the platform gdt was compiled but rather on my processor settings...

From the experiment I still got the same result. My structs are still 4 bytes. Changing align settings in the struct did not solve the issue. Changing the default packing did change the outcome...

2. From your comment I decided to test if the bitfields could be misinterpreted because of the builtin types size... If I change my types to `ushort` instead of unsinged, the size of the struct is represented correctly...

This seems related to structure packing being not entirely consistent with compiler behaviour... From your comments, maybe not only related to Xtensa but also to other platforms maybe...

For the time being I will change my source code to use shorter builtin type definitions but I guess that it would be interesing for maintainters to have a look at this. To my eyes I think this is a bug but I don't really know if they would consider otherwise...

Thanks for the feedback @Wall-AF! :)

**Comment by antoniovazquezblanco** [comment]
After some more testing, this also happens with enums. Now I doubt if the CParser interprets the packed attribute...

The following snippet compiles correctly with the Xtensa toolchain:
```c
typedef enum __attribute__((packed))
{
    A = 0x00,
    B = 0x01,
} enum_t;
static_assert(sizeof(enum_t) == 1);
```

But in ghidra it is shown as a 4 byte enum...

**Comment by antoniovazquezblanco** [comment]
At least, for enums, it seems there is a pending TODO:
https://github.com/NationalSecurityAgency/ghidra/blob/7bb5e0eb16e37abf488b8ee322af99bf6f1ee360/Ghidra/Features/Base/src/main/javacc/ghidra/app/util/cparser/C/C.jj#L523C83-L523C100

**Comment by ghidra1** [comment]
Unfortunately, this has been a known issue since the introduction of bitfields.  The problems stems from our Union implementations which only support a zero component offset combined with the packing mechanism which uses the smallest possible unit size for bitfield which would require a non-zero component offset.  Due to our limited support for bitfields at this point, particularly in the context of a union, this issue will likely remain unresolved for quite some time.

**Comment by ghidra1** [comment]
Sorry, at first glance I thought you were defining the bitfields within a union, which is permitted but is where we have trouble.  You may want to change the pack size of your structure to pack=1 instead of default.  This may address your issue.

**Comment by antoniovazquezblanco** [comment]
Thank you for the feedback @ghidra1. I've identified two issues with the parser. I decided to take it one step at a time.

For the time being I tackled the enum packing issue in https://github.com/NationalSecurityAgency/ghidra/pull/6780.

I will look into how CParser parses the `__attribute__((packed))` attribute in structs to see if there is any issue when I find some spare time. For the time being I've changed my source code to use smaller types and that won't need changing the default packing settings.

As you mention, this is not related to bitfield unions and also not related to the Xtensa architecture also. Sorry for the prompt issue post with little to no info. I hope the pull request will help compensate for some of the time wasted :D
